lib/inventory/inventory_api.rb

```diff
@@ -56,15 +54,13 @@ def list_resource_types(feed = nil)
         the_feed = hawk_escape feed
         ret = http_get('/feeds/' + the_feed + '/resourceTypes')
       end
-      val = []
-      ret.each { |rt| val.push(ResourceType.new(rt)) }
-      val
+      ret.map { |rt| ResourceType.new(rt) }
     end
 
     # List the resources for the passed feed and resource type. The representation for
     # resources under a feed are sparse and additional data must be retrived separately.
     # It is possible though to also obtain runtime properties by setting #fetch_properties to true.
-    # @param [String] feed The id of the feed the type lives under. Can be nil for feedless types
+    # @param [String] feed The id of the feed the type lives under. Can be nil for all feeds
```

Comment:
Right, I have the same understanding. Feedless ~ without feed (directly under environment) vs under feed. The reason, I've changed this is that
```bash
curl -u jdoe:password http://localhost:8080/hawkular/inventory/resourceTypes
```
actually fetches bot resource types that are reported under the feed (WF, OS, ..) and without the feed (just URL)

There is also a query param for fethcing only the feedless ones `?feedless=true`